ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/parameters/JsonCallParameter.java

```diff
@@ -15,44 +15,36 @@
 package org.hyperledger.besu.ethereum.api.jsonrpc.internal.parameters;
 
 import org.hyperledger.besu.ethereum.core.Address;
+import org.hyperledger.besu.ethereum.core.Gas;
 import org.hyperledger.besu.ethereum.core.Wei;
 import org.hyperledger.besu.ethereum.transaction.CallParameter;
 
+import java.util.Optional;
+
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import com.google.common.annotations.VisibleForTesting;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import org.apache.tuweni.bytes.Bytes;
 
 public class JsonCallParameter extends CallParameter {
```

Comment:
Why can't we just use the `CallParameter` with all of these annotations? Why do we need a separate `JsonCallParameter` class?

Reply:
This was what I wanted to do at the beginning but the `CallParameter` class is in the `core` module and allows to use the `TransactionSimulator`. I looked in the `core` module and outside of the tests no class has this kind of annotation


`CallParamater` is used in code to call the` TransactionSimulator` and `JsonCallParamater` is used to call RPC APIs. You may want to pass additional fields to the API without needing to pass it to the simulator. So for me it is two distinct objects

Reply:
I can only easily imagine the simulator taking _more_ parameters than a call parameter in the rpc since the simulator would need to know at least as much info as a normal call to do its job. Even if that happens eventually, that person will be empowered to split it in whichever way works for them.

I don't think that other classes in the package not using `JsonProperty` annotations need block us from adding it. If it keeps the data model cleaner that's a plus. Also, if "it hasn't been done before" becomes a justification for things then many people can come across the same problem with the same solution and discard it because "it hasn't been done before", thus preventing others from seeing it has been done before.

Reply:
>I don't think that other classes in the package not using JsonProperty annotations need block us from adding it.

I just merged the two classes. I don't have a strong opinion on that.

>Even if that happens eventually, that person will be empowered to split it in whichever way works for them.

👍